crates/stack/src/zigbee_stack.rs

```diff
@@ -36,13 +37,20 @@ mod route;
 // The number of the most recent samples taken into consideration SHOULD be n = 3, which
 // eliminates single outliers maintains a fast response to real changes in link quality,
 // and keeps memory requirements to a minimum.
-const LINK_QUALITY_SAMPLES: usize = 3; // For simplicity, keep this odd
+const LINK_QUALITY_SAMPLES: usize = 3;
 
+/// Compute the link cost (1-7) based on the LQI (0-255).
 fn lqi_to_link_cost(lqi: u8) -> u8 {
-    // TODO: is a linear mapping good enough?
-
-    // Remap 0-255 to 1-7
-    ((1.0 - (lqi as f32) / 255.0) * 6.0 + 1.0).round() as u8
+    match lqi {
+        0..=16 => 7,
+        17..=32 => 6,
+        33..=64 => 5,
+        65..=96 => 4,
+        97..=128 => 3,
+        129..=192 => 2,
+        193..=255 => 1,
+        // 0 corresponds to "unknown LQI"
```

Comment:
this could return an option instead when the LQI is zero. We could also make the return type a newtype: `LinkCost` to enhance readability further.

Reply:
Any valid LQI will map to a link cost of 1-7, it's just that the spec uses `0` to mean "unknown" link costs. I think we can deviate from the spec here and make it an `Option` as you said.